bin/hbase-config.sh

```diff
@@ -168,3 +168,27 @@ if [ -z "$JAVA_HOME" ]; then
 EOF
     exit 1
 fi
+
+function read_java_version() {
+  properties="$("${JAVA_HOME}/bin/java" -XshowSettings:properties -version 2>&1)"
+  #shellcheck disable=SC2016 # shellcheck cannot see through "${AWK}"
```

Comment:
nit, but this usually means we didn't tell shellcheck how to source some file we presume is present in the environment. in this case hbase-env. it'd be nice if we fixed that, but I don't think it needs to be your problem in this PR.

Reply:
Oh really?

I think there's more to it than that. I added on this patch but I still get the SC2016 warning.

```
modified   bin/hbase-config.sh
@@ -130,6 +130,7 @@ fi
 # Source the hbase-env.sh.  Will have JAVA_HOME defined.
 # HBASE-7817 - Source the hbase-env.sh only if it has not already been done. HBASE_ENV_INIT keeps track of it.
 if [ -z "$HBASE_ENV_INIT" ] && [ -f "${HBASE_CONF_DIR}/hbase-env.sh" ]; then
+  # shellcheck source=../conf/hbase-env.sh
   . "${HBASE_CONF_DIR}/hbase-env.sh"
   export HBASE_ENV_INIT="true"
 fi
@@ -171,7 +172,6 @@ fi
 
 function read_java_version() {
   properties="$("${JAVA_HOME}/bin/java" -XshowSettings:properties -version 2>&1)"
-  #shellcheck disable=SC2016 # shellcheck cannot see through "${AWK}"
   echo "${properties}" | "${GREP}" java.runtime.version | head -1 | "${AWK}" 'BEGIN {FS = " = "} ; {print $NF}'
 }
 
```

Reply:
Oh I should read closer. SC2016 isn't about being able to turn "${AWK}" into the contents of that variable. it's about the expression being passed to awk. Specifically the "print $NF" that we want AWK to handle and not the shell.

If we want to stick with AWK I think it's fine to disable the check. I'd find the comment easier to understand if it was phrased like "we mean to have AWK handle the variable resolution inside the passed awk script"

Otherwise you could get the same output using sed without anything that could be mistaken for a bash variable

```
   echo "${properties}" | "${GREP}" java.runtime.version | head -1 | "${SED}" -E -e 's/.* = ([^ ]*)/\1/'
```

Reply:
I'm fine with using `sed`, it would be `"${SED}"`. Is `-E` portable enough for our needs?

Reply:
I don't know off hand how portable `-E` is. In the example I only used it to avoid escaping the parentheses, so you can drop it and change the expression to `'s/.* = \([^ ]*\) */\1/'` (I forgot to strip trialing spaces in the original example)

Reply:
Switching to sed.